src/golang/cmd/server/server/aqueduct_server.go

```diff
@@ -298,3 +310,21 @@ func IndexHandler() func(w http.ResponseWriter, r *http.Request) {
 
 	return http.HandlerFunc(fn)
 }
+
+// Pause puts the server in system maintenance mode by blocking all new requests
+// and waits for all active requests to finish.
+// It is the responsibility of the caller to call s.Restart() to allow requests
+// to be processed again once the system maintenance is complete.
+func (s *AqServer) Pause() {
```

Comment:
how do we know that all active requests are finished?

Reply:
Since each request holds a read lock for the duration of the request (https://github.com/aqueducthq/aqueduct/pull/624/files#diff-d54562197a8af971fc47b82fe5c116481a4ac0b4de1f360070a943d1b06f19aaR53), we know for a fact that all requests are finished if this method can acquire an exclusive lock.